app/api/v2/managers/operation_api_manager.py

```diff
@@ -20,3 +48,60 @@ async def get_operation_object(self, operation_id: str, access: dict):
         if operation.match(access):
             return operation
         raise JsonHttpForbidden(f'Cannot view operation due to insufficient permissions: {operation_id}')
+
+    async def setup_operation(self, data: dict, access: BaseWorld.Access):
+        """Applies default settings to an operation if data is missing."""
+        planner_name = data.pop('planner', {}).get('name', '')
+        planner = await self._construct_planner(planner_name)
+        adversary_data = data.pop('adversary', {})
+        adversary_id = adversary_data.get('adversary_id', '')
+        adversary = await self._construct_adversary(adversary_id)
+        group_data = data.pop('host_group', '')
+        agents = await self._construct_agents(group_data)
```

Comment:
I'm wondering if we can just run `update_operation` after creating the operation and just have that method take care of populating agents, since it'll pull in appropriate agents based on the operation group